2019-10-24  Rui Deng  <classrui@lnxsrv07.seas.ucla.edu>

        * timespec.h (timespec_comp): Changed comparison mechanism from subtrac		tion to	direct comparison to avoid overflow. This function now returns         1, -1, 0 if A>B, A<B, A==B, respectively.
--- timespec.h  2005-09-21 23:05:39.000000000 -0700
+++ new_timespec.h   2019-10-24 20:49:50.794767000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  else if (a.tv_sec < b.tv_sec)
+    return -1;
+  return 0;
 }
